### PR TITLE
fix: Windows-rendering of content

### PR DESCRIPTION
**Pull request by bmuenzenmeyer**
## Description

I'd hadn't clicked around the site locally on my Windows machine in a bit, instead only rendering the homepage. The homepage was rendering fine, but every other page was 404ing content due to Windows pathing. I fixed this in two cases, indexes and deeper paths.

image

## Validation

All of these pages should continue to render for you locally.

- http://localhost:3000/en
- http://localhost:3000/en/about/
- http://localhost:3000/en/about/governance

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: next.dynamic.mjs [hunk 1]
```diff
@@ -41,10 +41,13 @@ const getAllPaths = async () => {
     (locale = '') =>
     (files = []) =>
       sourcePages.map(filename => {
-        const path = filename.replace(nextConstants.MD_EXTENSION_REGEX, '');
-
+        let path = filename.replace(nextConstants.MD_EXTENSION_REGEX, '');
```
Comment (ovflowd) [inline next.dynamic.mjs]: let's call this pathname and also add an inline comment of what this replace regex does

File: next.dynamic.mjs [hunk 2]
```diff
@@ -41,10 +41,13 @@ const getAllPaths = async () => {
...
+        // remove trailing slash for correct Windows pathing of the index files
+        if (path.length > 1 && path.endsWith(sep)) {
+          path = path.substring(0, path.length - 1);
+        }
```
Comment (ovflowd) [inline next.dynamic.mjs:49]:
```suggestion
        }

```

**Review by ovflowd — APPROVED** [review]
Let's ensure on Linux and macOS; this is working as expected.

We probably in the future should unit test this with examples of how each OS would handle paths.

**Review by aymen94 — APPROVED** [review]

**Review by mikeesto — APPROVED** [review]
All seems OK on MacOS

**Pull request merged**